Explain PostgreSQL limitations (#974)
cap-js/docs: guides/databases-postgres.md

## Provisioning a DB Instance

To connect to a PostgreSQL offering from the cloud provider in Production, leverage the [PostgreSQL on SAP BTP, hyperscaler option](https://discovery-center.cloud.sap/serviceCatalog/postgresql-hyperscaler-option).

<div markdown="1" class="impl java">

There are some limitations on the BTP to consume a PostgreSQL instance from a CAP Java application:

> - Only the Java buildpack `java_buildpack` provided by the Cloudfoundry community allows to consume a PostgreSQL service from a CAP Java application. To use this buildpack, configure it in the service module section of your mta.yaml

Reviewer: ```suggestion
- Only the Java buildpack `java_buildpack` provided by the Cloud Foundry community allows to consume a PostgreSQL service from a CAP Java application. To use this buildpack, configure it in the service module section of your mta.yaml
```

> - By default, the `java_buildpack` initializes the PostgreSQL datasource with the Java CFEnv library, but it's required to let the CAP Java runtime initialize the PostgreSQL datasource. This can be achieved by setting the environment variable `CFENV_SERVICE_<PG_SERVICE_NAME>_ENABLED` to `false` at your CAP Java service module. The placeholder `<PG_SERVICE_NAME>` needs to be replaced with the real service instance name of your PostgreSQL database.

Reviewer 2: maybe

```suggestion
- By default, the `java_buildpack` initializes a PostgreSQL datasource with the Java CFEnv library. However, to work properly with CAP, the PostgreSQL datasource must be created by the CAP Java runtime and not be the buildpack. You need to disable the datasource initialization by the buildback by setting the environment variable `CFENV_SERVICE_<PG_SERVICE_NAME>_ENABLED` to `false` at your CAP Java service module. The placeholder `<PG_SERVICE_NAME>` needs to be replaced with the real service instance name of your PostgreSQL database.
```

> - By default, the `java_buildpack` initializes the PostgreSQL datasource with the Java CFEnv library, but it's required to let the CAP Java runtime to initialize the PostgreSQL datasource. This can be achieved by setting the environment variable `CFENV_SERVICE_<PG_SERVICE_NAME>_ENABLED` to `false` at your CAP Java service module. The placeholder `<PG_SERVICE_NAME>` needs to be replaced with the real service instance name of your PostgreSQL database.

Reviewer: ```suggestion
- By default, the `java_buildpack` initializes the PostgreSQL datasource with the Java CFEnv library, but it's required to let the CAP Java runtime  initialize the PostgreSQL datasource. This can be achieved by setting the environment variable `CFENV_SERVICE_<PG_SERVICE_NAME>_ENABLED` to `false` at your CAP Java service module. The placeholder `<PG_SERVICE_NAME>` needs to be replaced with the real service instance name of your PostgreSQL database.
```

The following example shows both configuration settings applied to a CAP Java service module in an mta.yaml:

```yaml 
modules:
  - name: bookshop-pg-srv
    type: java
    path: srv
    parameters:
      buildpack: java_buildpack
    properties:
        SPRING_PROFILES_ACTIVE: cloud
        JBP_CONFIG_COMPONENTS: '{jres: ["JavaBuildpack::Jre::SapMachineJRE"]}'

>         JBP_CONFIG_SAP_MACHINE_JRE: '{ jre: { version: "17.+" } }'

Reviewer 2: or 21?

Author: will test if this will also work with the community buildpack for Java, not tested yet.

<div markdown="1" class="impl java">

To consume a PostgreSQL instance from a CAP Java application running on SAP BTP, consider the following:

- Only the Java buildpack `java_buildpack` provided by the Cloud Foundry community allows to consume a PostgreSQL service from a CAP Java application.

- By default, the `java_buildpack` initializes a PostgreSQL datasource with the Java CFEnv library. However, to work properly with CAP, the PostgreSQL datasource must be created by the CAP Java runtime and not by the buildpack. 

The following example shows these configuration settings applied to a CAP Java service:

::: code-group
```yaml [mta.yaml]
modules:
  - name: bookshop-pg-srv
    type: java
    path: srv
    parameters:
      buildpack: java_buildpack
    properties:
        SPRING_PROFILES_ACTIVE: cloud
        JBP_CONFIG_COMPONENTS: '{jres: ["JavaBuildpack::Jre::SapMachineJRE"]}'
        JBP_CONFIG_SAP_MACHINE_JRE: '{ jre: { version: "17.+" } }'

>         # We do not want cfenv to configure the DataSource for us
>         # cfenv uses names of the services, so this variable must be adapted if needed
>         # as CFENV_SERVICE_[pg-service-name]_ENABLED
>         # See https://docs.cloudfoundry.org/buildpacks/java/configuring-service-connections.html

Reviewer 3: I think we can even remove this as I feel this is too long. Then people can rather read the following note (which I moved here from the bulleted list).
```suggestion

```

- Only the Java buildpack `java_buildpack` provided by the Cloud Foundry community allows to consume a PostgreSQL service from a CAP Java application. To use this buildpack, configure it in the service module section of your mta.yaml

> - By default, the `java_buildpack` initializes a PostgreSQL datasource with the Java CFEnv library. However, to work properly with CAP, the PostgreSQL datasource must be created by the CAP Java runtime and not be the buildpack. You need to disable the datasource initialization by the buildback by setting the environment variable `CFENV_SERVICE_<PG_SERVICE_NAME>_ENABLED` to `false` at your CAP Java service module. The placeholder `<PG_SERVICE_NAME>` needs to be replaced with the real service instance name of your PostgreSQL database.

Reviewer 2: ```suggestion
- By default, the `java_buildpack` initializes a PostgreSQL datasource with the Java CFEnv library. However, to work properly with CAP, the PostgreSQL datasource must be created by the CAP Java runtime and not by the buildpack. You need to disable the datasource initialization by the buildback by setting the environment variable `CFENV_SERVICE_<PG_SERVICE_NAME>_ENABLED` to `false` at your CAP Java service module. The placeholder `<PG_SERVICE_NAME>` needs to be replaced with the real service instance name of your PostgreSQL database.
```
